src/config.rs

```diff
@@ -425,29 +425,79 @@ pub struct HtmlConfig {
     pub livereload_url: Option<String>,
     /// Should section labels be rendered?
     pub no_section_label: bool,
+    /// Search settings. If `None`, the default will be used.
```

Comment:
> If `None`, the default will be used.

I believe this is already sorted out by the `#[serde(default)]` annotation on `HtmlConfig`. That annotation tells serde to use `Search::default()` if it can't deserialize the settings.

To me, when I see `Option<Search>` I'd think this indicates that searching is optional and when no `Search` is found in my `book.toml` this means my book won't have a search bar... Am I interpreting it wrong, or is this comment out of sync with the implementation?

Reply:
> I believe this is already sorted out by the #[serde(default)] annotation on HtmlConfig. That annotation tells serde to use Search::default() if it can't deserialize the settings.

The default it'll use is `None`, since that's how `Default::default::<Option>()` is implemented: https://doc.rust-lang.org/src/core/option.rs.html#900-906, and we're using `#[derive(Default)]`. The html renderer will then manually use `unwrap_or_default` to get the default configuration. The reason to do this is so we can print the warning message if the user is using mdbook compiled without `feature = "search"`, but does have an `[output.html.search]` table in their `book.toml`.

Reply:
It may be less confusing overall to just bring back the `[output.html.search.enabled]` field.